vimeo/vimeo-networking-java
# VA-1367 Gson Type Adapters for concrete requests

#### Ticket
[VA-1367](https://vimean.atlassian.net/browse/VA-1367)

#### Ticket Summary
Adding in Gson TypeAdapters for concrete requests (PictureResource, Comment, VimeoAccount, and Video) and all objects those depend on.... which is most objects.

#### Implementation Summary
I added type adapters, and used the TypeAdapterFactory to provide TypeAdapters with other TypeAdapters to handle nested objects.
I introduced a simple logging mechanism that will use the VimeoClient logger and fallback to system.out if VimeoClient hasn't been initialized. Since operating in the context of models, this is a possibility we need to account for.

#### How to Test
Run through the app (see VimeoAndroid PR) and verify that functionality still works. You can put breakpoints in the TypeAdapter reads (and writes) to see that they are getting used. To test easily, you can add some hacky code to use gson fromJson and toJson methods. 

@howardrigberg @anthonycr Please feel free to review as well.
